iaso/migrations/0264_data_delete_change_requests_duplicates_uuid.py

```diff
@@ -0,0 +1,33 @@
+# Generated by Django 4.2.9 on 2024-03-08 09:37
+
+from django.db import migrations
+from django.db.models import Count
+
+
+def migrate_data_forward(apps, schema_editor):
+    """
+    This data migration deletes `OrgUnitChangeRequest` duplicates sharing the same uuid.
+    It's a prerequisite for adding a unique constraint on `OrgUnitChangeRequest.uuid`.
+    """
+    OrgUnitChangeRequest = apps.get_model("iaso", "OrgUnitChangeRequest")
+
+    duplicated_uuids = (
+        OrgUnitChangeRequest.objects.values_list("uuid", flat=True)
+        .annotate(Count("uuid"))
+        .order_by()
+        .filter(uuid__count__gt=1)
+    )
+
+    for uuid in duplicated_uuids:
+        # Retrieve change requests sharing the same UUID except for the oldest one.
+        change_requests_to_delete = OrgUnitChangeRequest.objects.filter(uuid=uuid).order_by("created_at")[1:]
+        for change_request in change_requests_to_delete:
+            change_request.delete()
```

Comment:
Should we not check that they have the same status? E.g: if one of the dupes has been approved we should probably delete the other

Reply:
But I don't know if there's a big risk/probability to have that issue in prod

Reply:
Good point. I investigated the data and I need to think a little bit more about how to handle duplicates.

Reply:
Fixed in https://github.com/BLSQ/iaso/pull/1173/commits/2ccdcbca626e49ab56f8ac78c322fee46d6052f4